src/main.ts

```diff
@@ -6,45 +6,59 @@ import {
   generateRowKey,
 } from "./core/structure/key/keyTypes.ts";
 import { CellInfo } from "./core/structure/sheet.types.ts";
-
+import { GenerateRowLabel } from "../utils/helpers.ts";
 export default class LightSheet {
   ui: UI;
   options: LightSheetOptions;
   sheet: Sheet;
   onCellChange?;
 
+  #defaultRowCount: number = 4;
+  #defaultColCount: number = 4;
+
   constructor(targetElement: Element, options: LightSheetOptions) {
     this.options = options;
     this.sheet = new Sheet();
     this.ui = new UI(
       targetElement,
       this,
-      this.options.data.length,
-      this.options.data[0].length,
+      this.options.data?.length ?? this.#defaultRowCount,
     );
-    this.initializeData();
+    this.#initializeData();
     if (options.onCellChange) {
       this.onCellChange = options.onCellChange;
     }
   }
 
-  initializeData() {
+  #initializeData() {
```

Comment:
Since this method is initializing the table view itself and works even if data isn't available, `initializeTable` or something similar would be a more descriptive name now.

Reply:
Yep Changed it to initializeTable